### PR TITLE
Wait for action response in message

### LINKED ISSUE #151: [Server]: Wait for robot to acknowledge completion of commands before continuing execution
#### Task Description:

Right now, we don't have a way of tracking robots status in executing a command are running. In certain instances, we need to know that a robot is done performing a move before we tell it to move again. This issue is to ensure that individual command promises do not resolve/reject until a given robot sends a response message acknowledging that it has completed or failed to complete a given command..

other TODOs related to this issue: 
- create issue for HW code to emit the success/fail message (@jasonappah / @vcinventerman )
- Document/refine bot command execution flow discussed with @democat3457 and @xXDMOGXx (currently in discord, needs to move to repo) - @jasonappah

### PR DESCRIPTION
**Pull request by democat3457**
Uses an EventEmitter to resolve Promises that indicate a bot command is completed. This is a core component of the command flow and ensures commands do not happen simultaneously while robots are moving.

Also, merges GameTypes between client- and game-type.

Fixes #151.

Tests need to be written to ensure this workflow happens accordingly:
1. Construct command group filled with dummy commands.
2. Mock robot sockets, and mock successful response after some time.
3. Check:
  - Ensure command group is added to the running command list
  - Ensure the commands run in the proper order
  - Ensure command does not return until successful response is sent
  - When successful response is sent, ensure command returns
  - When commands have returned, ensure command group finishes and removes itself from running command list

Also test the requirement system: ensure that when two commands that require the same robot are executed simultaneously, an exception is thrown.

### CODE REVIEW

**Review by jasonappah — COMMENTED** [review]
looks good to me, but would like to wait for input from @AlexKempen and maybe a little more sleep 😄

File: src/server/api/tcp-interface.ts [hunk 1]
```diff
@@ -146,6 +161,15 @@ export class BotTunnel {
         console.log({ msg });
         this.socket.write(msg);
     }
+
+    async waitForActionResponse(): Promise<void> {
+        return new Promise((res, rej) => {
+            this.emitter.once("actionComplete", (args) => {
+                if (args.success) res();
+                else rej(args.reason);
```
Comment (AlexKempen) [inline src/server/api/tcp-interface.ts:169]: I think we should always use brackets for readability. I'd imagine there's also an eslint rule that could do it automatically?

**Review by AlexKempen — COMMENTED** [review]
Looks pretty good to me, some minor nit comments. In the future we need to have a more complete talk about what error cases we want and how to handle them - we'll also need to hash out an error recovery pipeline. But that shouldn't be strictly necessary for the demo.

**Review by AlexKempen — APPROVED** [review]
Looks good to me.

**Pull request merged**